### ISSUE TITLE
Problems with xboa migrate

### ISSUE THREAD

**Issue by ar-jan** [omega8cc/boa#1146]
I did a migration following https://github.com/omega8cc/boa/blob/master/docs/MIGRATE.txt today, and I encountered some problems. Since as far as I can see I followed all instructions and don't have any config/setup that's out of the ordinary, I think there may be some issues in the procedure. I'm listing all the information that looks relevant, please let me know if I should provide further details.

- Both source and target server running BOA 3.2.0.
- Both servers have PHP 5.6 and 7.0 configured and installed, with 5.6 as default (though the old system has previously also used 5.5).
- I'm using Let's Encrypt for most sites.

I followed the documented steps, and the output of all commands looked as expected, on both source and target system, except for the following during import on the target system. For every site the same fatal error happened (twice):

> Dom is www.example.com
> site www.example.com raw _DBN is examplecom_0
> site www.example.com clean _DBN is examplecom_0
> site www.example.com raw _DBP is abcDEf
> site www.example.com clean _DBP is abcDEf
> INFO: site www.example.com db_host on o1 fixed
> INFO: site www.example.com db setup on o1 complete
> INFO: site www.example.com db import on o1 complete
> Fatal error: Call to a member function get_services() on null in /data/disk/o1/.drush/sys/provision/Provision/Context.php on line 298
> Fatal error: Call to a member function get_services() on null in /data/disk/o1/.drush/sys/provision/Provision/Context.php on line 298
> INFO: site www.example.com verify on o1 scheduled
> Done import for /data/disk/o1
> INFO: import for o1 to 123.456.789.123  complete

Despite the error, it seems the sites are imported, so I followed the rest of the steps. After completing the last step, I got "502 Bad Gateway" for all of the sites hosted.

The /var/log/php/*.log files on the target system showed no errors, other than the "Call to a member function get_services()" which had appeared during import.
It seems that just an extra `service nginx reload` on the target system fixed the problem (I had already done the nginx reload before as instructed). The error looks like this one https://www.drupal.org/node/2718437, but that problem was already fixed long ago.

So now the sites were working, and I pointed a few domains to the new IP address. But in the new control panel, the site verification tasks were not working. The problematic parts from the log, site verify tasks on the target server:

> Could not find provision alias named: @platform_Drupal7511Prod
> Drush command terminated abnormally due to an unrecoverable error. Error: Call to a member function get_services() on null in /data/disk/o1/.drush/sys/provision/Provision/Context.php, line 298	
> The external command could not be executed due to an application error.
> [...]
> SSL certificate directory for example.com permissions of /data/disk/o1/config/ssl.d/example.com could not be changed to 700. (chmod to 700 failed on /data/disk/o1/config/ssl.d/example.com)
> SSL certificate directory for example.com path /data/disk/o1/config/ssl.d/example.com is not writable.
> [...]
> failed to generate SSL key in /data/disk/o1/config/ssl.d/example.com/openssl.key	
> failed to generate signing request for certificate in /data/disk/o1/config/ssl.d/example.com/openssl.csr	
> failed to generate self-signed certificate in /data/disk/o1/config/ssl.d/example.com/openssl.crt
> [...]
> SSL Certificate directory for example.com on web01.example.org permissions of /data/disk/o1/config/server_master/ssl.d/example.com could not be changed to 700. (chmod to 700 failed on /data/disk/o1/config/server_master/ssl.d/example.com)
> SSL Certificate directory for example.com on web01.example.org path /data/disk/o1/config/server_master/ssl.d/example.com is not writable.

All the above problems seem to be caused by permission problems - the files in these dirs are owned by `www55:users`, while on the old system they were `o1:users`.

After changing with as follows, the site verifications are also working:
`chown -R o1:users /data/disk/o1/.drush/ /data/disk/o1/config/ssl.d/ /data/disk/o1/config/server_master/ssl.d/ /data/disk/o1/tools/le/certs/ /data/disk/o1/clients/ /data/disk/o1/tools/le`.

**Comment by omega8cc** [comment]
Thanks for the detailed bug report! We love to receive such helpful reports, even if we hate to see confirmation that we have introduced regressions..

I'm afraid it could be related to these changes we have made without actually testing the impact on this script: https://github.com/omega8cc/boa/commit/80a2bcb1c6fba5c86f215ea80fa31ebd2c9b9670

The problem is that BOA auto-updates the `xboa` script using BOA head version.

It was based on our local migrations results, where adding `--numeric-ids` solved critical issues.

But here clearly something went wrong.
